tests/conftest.py

```diff
@@ -252,29 +252,37 @@ def pytest_generate_tests(metafunc):
 
 
 def pytest_collection_modifyitems(config, items):
-    if config.getoption("--aws-integration"):
-        # --aws-integration given in cli: do not skip aws-integration tests
-        return
-    if config.getoption("--docs-tests"):
-        # --docs-tests given in cli: do not skip documentation integration tests
-        return
-    if config.getoption("--cloud"):
-        # --cloud given in cli: do not skip GX Cloud integration tests
-        return
-
-    skip_aws_integration = pytest.mark.skip(
-        reason="need --aws-integration option to run"
+    @dataclass
+    class Category:
+        mark: str
+        flag: str
+        reason: str
+
+    categories = (
+        Category(
+            mark="aws_integration",
+            flag="--aws-integration",
+            reason="need --aws-integration option to run",
+        ),
+        Category(
+            mark="docs",
+            flag="--docs-tests",
+            reason="need --docs-tests option to run",
+        ),
+        Category(mark="cloud", flag="--cloud", reason="need --cloud option to run"),
     )
-    skip_docs_integration = pytest.mark.skip(reason="need --docs-tests option to run")
-    skip_cloud = pytest.mark.skip(reason="need --cloud option to run")
 
-    for item in items:
-        if "aws_integration" in item.keywords:
-            item.add_marker(skip_aws_integration)
-        if "docs" in item.keywords:
-            item.add_marker(skip_docs_integration)
-        if "cloud" in item.keywords:
-            item.add_marker(skip_cloud)
+    for category in categories:
+        # If flag is provided, exit early so we don't add `pytest.mark.skip`
+        if config.getoption(category.flag):
```

Comment:
Not blocking: You traverse the list of tests for each flag that is set. The list of test is >> than the list of categories. Alternatives:

* You could skip this continue and for each test, check if it's in each category. 
* You could do through the categories first, collect the list of set ones and then go through the tests once and check for the set categories.

I don't think we need to worry about this too much, just wanted to point it out since it seems like you are doing some optimization here.